lib/api/controllers/authController.js

```diff
@@ -163,10 +197,121 @@ class AuthController {
   /**
    * List authentication strategies
    *
-   * @returns {Promise}
+   * @returns {Promise.<string[]>}
    */
   getStrategies() {
-    return this.kuzzle.pluginsManager.listStrategies();
+    return Bluebird.resolve(this.kuzzle.pluginsManager.listStrategies());
+  }
+
+  /**
+   * @param {Request} request
+   * @returns {Promise.<Object>}
+   */
+  createMyCredentials(request) {
+    let
+      createMethod,
+      validateMethod;
+
+    assertIsConnected(this.kuzzle, request);
+    assertHasStrategy(request);
+    assertIsStrategyRegistered(this.kuzzle, request);
+    assertHasBody(request);
+
+
+    createMethod = this.kuzzle.pluginsManager.getStrategyMethod(request.input.args.strategy, 'create');
```

Comment:
Same as @scottinet, mandatory functions are checked and sandboxed during the plugin registration. The only exception is getInfo that is not mandatory.